freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/instance/DefaultInstanceGroupProvider.java

```diff
@@ -55,6 +56,11 @@ public Template createDefaultTemplate(CloudPlatform cloudPlatform, String accoun
                     entry(AwsInstanceTemplate.EBS_ENCRYPTION_ENABLED, Boolean.TRUE),
                     entry(InstanceTemplate.VOLUME_ENCRYPTION_KEY_TYPE, EncryptionType.DEFAULT.name()))));
         }
+        if (cloudPlatform == CloudPlatform.AZURE && entitlementService.isAzureDiskSSEWithCMKEnabled(accountId)) {
+            template.setAttributes(new Json(Map.<String, Object>ofEntries(
+                    entry(AzureInstanceTemplate.DISK_ENCRYPTION_SET_ID, null),
+                    entry(AzureInstanceTemplate.MANAGED_DISK_ENCRYPTION_WITH_CUSTOM_KEY_ENABLED, Boolean.TRUE))));
+        }
```

Comment:
This combination is not fortunate:
* Setting `MANAGED_DISK_ENCRYPTION_WITH_CUSTOM_KEY_ENABLED` requires a valid `DISK_ENCRYPTION_SET_ID`, for the Azure logic cannot provision DES on-the-fly. Please do the same as in `InstanceTemplateRequestToTemplateConverter`.
* Checking for the presence of the entitlement makes no sense; it is only a pre-requisite for using Disk SSE with CMK but does not imply that the feature has been actually requested during environment provisioning. The latter depends entirely on whether `diskEncryptionSetId` has been specified in the environment creation request or not.